esmvaltool/recipes/recipe_rainfarm.yml

```diff
@@ -0,0 +1,49 @@
+# recipe_rainfarm.yml
+---
+
+documentation:
+  description: |
+    Recipe for performing stochastic downscaling of precipitation fields
+    calling the RainFARM package by J. von Hardenberg (ISAC-CNR)
+  authors:
+    - arno_en
+    - hard_jo
+
+  references:
+    - donofrio14jh
+    - rebora06jhm
+
+  projects:
+    - c3s-magic
+
+datasets:
+  - {dataset: EC-Earth,       project: CMIP5, exp: historical, ensemble: r8i1p1, start_year: 1997, end_year: 1997}
```

Comment:
This is causing problem, since on my drs `EC-Earth` is stored as `EC-EARTH`.
You have reported this problem in #330, I thought it was solved, but it is not.
@valeriupredoi any suggestion?

Reply:
@mattiarighi what are the exact problems - confusion between EC-EARTH and EC-Earth in datafinder?

Reply:
Right. Apparently `_data_finder.py` is case sensitive.

Reply:
@mattiarighi I thought the issue was already solved by the new pre-proc as @valeriupredoi mentioned in #330 . I have both EC-Earth and EC-EARTH, the pre-proc should correctly halt if it doesn't find the exact match? Of course we may want to type EC-EARTH in the example!

Reply:
@mattiarighi - shall I change the example to EC-EARTH and pull request again?

Reply:
now I'm thoroughly confused: are you looking for EC-Earth and data finder finds both EC-Earth __and__ EC-EARTH? it shouldn't, datafinder uses `files = fnmatch.filter(files, filename)` which is case-sensitive

Reply:
Yes, indeed the preprocessor returns an error since it cannot find the file.

Ideally the data_finder should be able to handle both, but I do not know whether this is easy to implement (also considering that dataset names are hard-coded in `config-developer.yml` in the function that retrieves the corresponding institutions).

Reply:
@valeriupredoi the data finder stops because it is looking for `EC-Earth` but I have `EC-EARTH` in my data pool (both for ETHZ and DKRZ).
The question is whether we want data_finder to behave like this or whether it should be more flexible.

Reply:
@earnone no need to pull request again. If you change it, commit and push, this PR will be updated automatically.

Reply:
`fnmatch` is case-sensitive (and no way you can turn that off) if the OS is case-sensitive (as most are); if we want to relax this we need to use `os.walk` probably, but the question is what are the risks of picking up unwanted patterns? I don't have an answer for this unfortunately :disappointed:

Reply:
OK, let's forget about it. :+1: 
@earnone if you could change the case in your recipes that would be great (also note my identical comment in #620).